annotated.go

```diff
@@ -582,31 +1059,193 @@ var _ Annotation = asAnnotation{}
+		} else {
+			outTypes = append(types, resType)
+		}
+		return outTypes, func(results []reflect.Value) []reflect.Value {
+			var (
+				outErr     error
+				outResults []reflect.Value
+			)
+			newOutResult := reflect.New(resType).Elem()
+
+			for i, r := range results {
+				if i == len(results)-1 && hasError {
+					// If hasError and this is the last item,
+					// we are guaranteed that this is an error
+					// object.
+					if err, _ := r.Interface().(error); err != nil {
+						outErr = err
+					}
+					continue
+				}
+				outResults = append(outResults, r)
+				if i == 0 {
+					for j := 1; j < resType.NumField(); j++ {
+						newOutResult.Field(j).Set(r.Field(j))
+					}
+				}
+			}
+
+			outResults = append(outResults, newOutResult)
+
+			if hasError {
+				if outErr != nil {
+					outResults = append(outResults, reflect.ValueOf(outErr))
+				} else {
+					outResults = append(outResults, _nilError)
+				}
+			}
+
+			return outResults
+		}, nil
+	}
+	// not yet went through a ResultTags annotation
+	for i, t := range types {
+		if isOut(t) || t == _typeOfError {
+			continue
+		}
+
+		field := reflect.StructField{
+			Name: fmt.Sprintf("Field%d", i),
+			Type: t,
+		}
+
+		if i < len(at.types) {
+			if !t.Implements(at.types[i]) {
+				return nil, nil, fmt.Errorf("invalid fx.As: %v does not implement %v", t, at.types[i])
+			}
+			field.Type = at.types[i]
+		}
+		fields = append(fields, field)
+	}
+	resType := reflect.StructOf(fields)
+	var outTypes []reflect.Type
+	if hasError {
+		outTypes = append(types[0:len(types)-1], resType, types[len(types)-1])
+	} else {
+		outTypes = append(types, resType)
+	}
+	return outTypes, func(results []reflect.Value) []reflect.Value {
+		var (
+			outErr     error
+			outResults []reflect.Value
+		)
+		newOutResult := reflect.New(resType).Elem()
+
+		for i, r := range results {
+			if i == len(results)-1 && hasError {
+				// If hasError and this is the last item,
+				// we are guaranteed that this is an error
+				// object.
+				if err, _ := r.Interface().(error); err != nil {
+					outErr = err
+				}
+				continue
+			}
+			outResults = append(outResults, r)
+			if !isOut(r.Type()) && newOutResult.NumField() > i+1 {
+				newOutResult.Field(i + 1).Set(r)
+			}
+		}
+
+		outResults = append(outResults, newOutResult)
+
+		if hasError {
+			if outErr != nil {
+				outResults = append(outResults, reflect.ValueOf(outErr))
+			} else {
+				outResults = append(outResults, _nilError)
+			}
+		}
+
+		return outResults
+	}, nil
+}
+
 type annotated struct {
-	Target     interface{}
-	ParamTags  []string
-	ResultTags []string
-	As         [][]reflect.Type
-	FuncPtr    uintptr
-	Hooks      []*lifecycleHookAnnotation
+	Target      interface{}
+	Annotations []Annotation
+	ParamTags   []string
+	ResultTags  []string
+	As          [][]reflect.Type
+	FuncPtr     uintptr
+	Hooks       []*lifecycleHookAnnotation
+	container   *dig.Container
```

Comment:
add a comment here explaining how this is used, or document it in the implementation part where it's using this container, and add a comment here that refers to that section.